File: sharedModule/src/main/java/ru/nekit/android/qls/setupWizard/ISetupWizardHolder.java
package ru.nekit.android.qls.setupWizard;

import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;

interface ISetupWizardHolder {

    BaseSetupWizard getSetupWizard();

    void showSetupWizardStep(ISetupWizardStep step, Object... params);

    Button getNextButton();

    Button getAltButton();

    boolean nextAction();

    void setUnconditionedNextAction(boolean value);

    void altAction();

    ViewGroup getToolContainer();

    View getView();
}

File: sharedModule/src/main/java/ru/nekit/android/qls/setupWizard/ISetupWizardHolder.kt
package ru.nekit.android.qls.setupWizard

import android.view.View
import android.view.ViewGroup
import android.widget.Button
import io.reactivex.Single

internal interface ISetupWizardHolder {

    val setupWizard: BaseSetupWizard

    fun getNextButton(): Button

    fun getAltButton(): Button

    fun getToolContainer(): ViewGroup

    fun getView(): View

    fun showSetupWizardStep(step: ISetupWizardStep, vararg params: Any)

    fun showNextSetupWizardStep()

    fun nextAction(): Single<Boolean>

    fun altAction()
}